extensions/metadata/sleepycat/src/main/java/org/exist/storage/md/MetaDataImpl.java

```diff
@@ -345,16 +366,27 @@ protected Meta getMeta(Metas doc, String key) {
 		return null;
 	}
 
-    public void streamMetas(DocumentImpl doc, MetaStreamListener listener) {
-        Metas metas = getMetas(doc);
+    public void streamMetas(XmldbURI uri, MetaStreamListener listener) {
+        Metas metas = getMetas(uri);
         if (metas == null)
             return;
+
+        HashSet<String> check = new HashSet<>();
```

Comment:
It's generally considered best practice and fits with the rest of the eXist codebase.